cmd/snap/cmd_changes.go

```diff
@@ -122,14 +125,57 @@ func (c *cmdChanges) Execute(args []string) error {
 
 func (c *cmdTasks) Execute([]string) error {
 	cli := Client()
-	return showChange(cli, c.Positional.ID)
+	var id changeID
+	if c.Positional.ID == "" && c.LastChangeType == "" {
+		return fmt.Errorf(i18n.G("please provide change ID or type with --last=<type>"))
+	}
+	if c.Positional.ID != "" && c.LastChangeType != "" {
+		return fmt.Errorf(i18n.G("change use ID and type together"))
+	}
+	if c.LastChangeType != "" {
+		kind := c.LastChangeType
+		// our internal change types use "-snap" postfix but let user skip it and use short form.
+		if kind == "refresh" || kind == "install" || kind == "remove" || kind == "connect" || kind == "disconnect" || kind == "configure" || kind == "try" {
+			kind += "-snap"
+		}
+		opts := client.ChangesOptions{
+			Selector: client.ChangesAll,
+		}
+		changes, err := cli.Changes(&opts)
+		if err != nil {
+			return err
+		}
+		if len(changes) == 0 {
+			return fmt.Errorf(i18n.G("no changes found"))
+		}
+		// sort by date, descending so that we will pick the most recent change of given kind.
+		sort.Sort(sort.Reverse(changesByTime(changes)))
+		chg := findFirstChangeByKind(changes, kind)
```

Comment:
Good idea. I was in a "functional programming" mindset doing doing that change ;). Done.